src/Directory.Build.props

```diff
@@ -35,6 +35,11 @@
     <CodeAnalysisRuleSet>$(MSBuildThisFileDirectory)CodeAnalysis.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(SolutionName)' == 'Sentry.Aot' and $(TargetFramework.StartsWith('net8'))">
+    <PublishAot>true</PublishAot>
```

Comment:
I'm not sure about defining this property as it may imply that we're publishing AOT libraries, which we're not. Instead, I'd leave it up to specific sample apps to specify the property as needed (and as supported by the current OS).

Reply:
yes, that makes sense.

Reply:
See the following log, specifically the sentry.dll compilation part. Warnings removed for brewity because this log is from my branch which doesn't fix the warnings yet.

```
dotnet publish ./samples/Sentry.Samples.Console.Basic -c release -f net8.0                
MSBuild version 17.8.0+6cdef4241 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
/usr/local/share/dotnet/sdk/8.0.100-rc.2.23502.2/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(311,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [/Users/ivan/dev/sentry-dotnet/samples/Sentry.Samples.Console.Basic/Sentry.Samples.Console.Basic.csproj::TargetFramework=net8.0]
  Sentry -> /Users/ivan/dev/sentry-dotnet/src/Sentry/bin/release/net6.0/Sentry.dll
  Sentry.Samples.Console.Basic -> /Users/ivan/dev/sentry-dotnet/samples/Sentry.Samples.Console.Basic/bin/release/net8.0/osx-arm64/Sentry.Samples.Console.Basic.dll
  Generating native code
....
  Sentry.Samples.Console.Basic -> /Users/ivan/dev/sentry-dotnet/samples/Sentry.Samples.Console.Basic/bin/release/net8.0/osx-arm64/publish/
```

Reply:
OK I'm not sure if using compiler directives to include/exclude stuff based on `#TRIMMABLE` will work at all. 

I tried ripping out the property group above and put this instead in `buildTransitive/Sentry.targets`:
```
<Target Name="SetTrimmableConstant" BeforeTargets="BeforeBuild" Condition="'$(PublishAot)' != 'true'">
  <PropertyGroup Condition="'$(PublishAot)' != 'true'">
    <DefineConstants>$(DefineConstants);TRIMMABLE</DefineConstants>
  </PropertyGroup>
</Target>
```

Then I packed that NuGet package and included it in a shinny new Console App (from a [local Nuget store](https://weblog.west-wind.com/posts/2022/Sep/11/Referencing-a-Local-Private-NuGet-Package-in-your-Solution)) with `<PublishAot>true</PublishAot>`. When I published that console app I got a whole bunch of trimming warnings from the Sentry.dll 😞 
 
We could, of course, take out all the `#TRIMMABLE` directives and just make those code paths the only code paths... i.e. Sentry is always trimmable. There are quite a few implications to that though:
- No enhanced stack traces
- No serializing anonymous/dynamic types in the Sentry protocol
- No automatic wiring up unhandled exception handlers for for WinUI applications
- Presumably a bunch more once we take trimming/aot beyond Sentry.csproj

🤔

Reply:
It should not be in buildTransitive. The way it was before in this PR made sense and should work. You'll get a sentry.dll with the compiler directive set correctly and then use that DLL to compile the final app

Reply:
I believe it's enough, for this PR, to concentrate on the sample apps included in this repo. Even just the console ones. If we get them to publish without warnings, we can make it work with nuget too.

Reply:
That makes sense. I'll resolve this conversation and create a new issue from the [comment below](https://github.com/getsentry/sentry-dotnet/pull/2732#discussion_r1368297025).

Reply:
Created https://github.com/getsentry/sentry-dotnet/issues/2759 for this.